### PR TITLE
[formrecognizer] Update error handling in samples

### PR DESCRIPTION
**Pull request by catalinaperalta**
Simplify error handling in samples

### CODE REVIEW

File: sdk/formrecognizer/azure-ai-formrecognizer/samples/v3.2/async_samples/sample_analyze_business_cards_async.py [hunk 1]
```diff
@@ -165,19 +165,17 @@ async def main():
...
         asyncio.run(main())
     except HttpResponseError as error:
+        print("For more information about troubleshooting errors, see the following guide: "
+              "https://aka.ms/azsdk/python/formrecognizer/troubleshooting")
         # Examples of how to check an HttpResponseError
         # Check by error code:
         if error.error is not None:
-            if error.error.code == "InvalidRequest":
-                print(f"Received an invalid request error: {error.error}")
-                sys.exit(1)
```
Comment (kristapratico) [inline sdk/formrecognizer/azure-ai-formrecognizer/samples/v3.2/async_samples/sample_analyze_business_cards_async.py:173]: Isn't this a pretty common error? Should we keep it?

File: sdk/formrecognizer/azure-ai-formrecognizer/samples/v3.2/async_samples/sample_analyze_business_cards_async.py [hunk 2]
```diff
@@ -165,19 +165,17 @@ async def main():
...
                 sys.exit(1)
         # If the inner error is None and then it is possible to check the message to get more information:
-        filter_msg = ["Generic error", "Timeout", "Invalid request", "InvalidImage"]
-        if any(example_error.casefold() in error.message.casefold() for example_error in filter_msg):
+        if "Invalid request".casefold() in error.message.casefold():
             print(f"Uh-oh! Something unexpected happened: {error}")
             sys.exit(1)
         # Print the full error content:
-        print(f"Full HttpResponseError: {error}")
+        print(f"Received a different error: {error}")
```
Comment (kristapratico) [inline sdk/formrecognizer/azure-ai-formrecognizer/samples/v3.2/async_samples/sample_analyze_business_cards_async.py:181]: I wonder if we can just wrap all these lines into the "Received a different error" group to simplify further. Seems the outcome is to print the {error} in either case.

File: sdk/formrecognizer/azure-ai-formrecognizer/samples/v3.2/async_samples/sample_analyze_business_cards_async.py (hunk 1 above)
Comment (catalinaperalta) [reply sdk/formrecognizer/azure-ai-formrecognizer/samples/v3.2/async_samples/sample_analyze_business_cards_async.py:173]: The way I decided to still showcase invalid request information was by checking the message for invalid request on line 177, kind of just showing options for how to access the error. If you think it's better to have invalid request here and some other type in the message check I can switch it up. Let me know!

File: sdk/formrecognizer/azure-ai-formrecognizer/samples/v3.2/async_samples/sample_analyze_business_cards_async.py (hunk 2 above)
Comment (catalinaperalta) [reply sdk/formrecognizer/azure-ai-formrecognizer/samples/v3.2/async_samples/sample_analyze_business_cards_async.py:181]: I was tempted to keep the check on line 177 to showcase that sometimes errors come in a different format. I could print the message specifically on line 178 to differentiate a bit more, thoughts?

File: sdk/formrecognizer/azure-ai-formrecognizer/samples/v3.2/async_samples/sample_analyze_business_cards_async.py (hunk 1 above)
Comment (kristapratico) [reply sdk/formrecognizer/azure-ai-formrecognizer/samples/v3.2/async_samples/sample_analyze_business_cards_async.py:173]: Okay, I just don't have a feel for the most common error codes returned. Seems we'd want to demonstrate capturing those if possible so we don't end up in the "else" case

File: sdk/formrecognizer/azure-ai-formrecognizer/samples/v3.2/async_samples/sample_analyze_business_cards_async.py (hunk 2 above)
Comment (kristapratico) [reply sdk/formrecognizer/azure-ai-formrecognizer/samples/v3.2/async_samples/sample_analyze_business_cards_async.py:181]: Ah I see. it's unfortunate there are two ways to parse the error, but I guess that's just how it is. just fyi, if it hits the "different error" it won't raise an exception /sys.exit. not sure if that was intentional.

**Review by kristapratico — APPROVED** [review]

File: sdk/formrecognizer/azure-ai-formrecognizer/samples/v3.2/async_samples/sample_analyze_business_cards_async.py (hunk 2 above)
Comment (catalinaperalta) [reply sdk/formrecognizer/azure-ai-formrecognizer/samples/v3.2/async_samples/sample_analyze_business_cards_async.py:181]: I did check and I saw it raising the error. I was relying on the programs sys.exit after the try except finished, but I'll add it for symmetry with the other checks.

File: sdk/formrecognizer/azure-ai-formrecognizer/samples/v3.2/async_samples/sample_analyze_business_cards_async.py (hunk 1 above)
Comment (catalinaperalta) [reply sdk/formrecognizer/azure-ai-formrecognizer/samples/v3.2/async_samples/sample_analyze_business_cards_async.py:173]: I'll add the check for the invalidrequest error code once again just to make sure we're catching all of those

File: sdk/formrecognizer/azure-ai-formrecognizer/samples/v3.2/async_samples/sample_analyze_business_cards_async.py (hunk 2 above)
Comment (catalinaperalta) [reply sdk/formrecognizer/azure-ai-formrecognizer/samples/v3.2/async_samples/sample_analyze_business_cards_async.py:181]: Nevermind, you're right it's not raising the error. I'll add that in

Comment (catalinaperalta) [reply sdk/formrecognizer/azure-ai-formrecognizer/samples/v3.2/async_samples/sample_analyze_business_cards_async.py]: Added it, thanks for pointing that out

**Pull request merged**